sdk/core/Azure.Core.Experimental/samples/Variant.md

```diff
@@ -0,0 +1,135 @@
+# Azure.Variant
+
+`Azure.Variant` is an implementation of a [tagged union](https://en.wikipedia.org/wiki/Tagged_union).  It can hold both reference and value types and can be used to avoid boxing .NET primitives.  The list of value types that Variant can hold without boxing is below.
+
+<details>
+<summary>Value types that `Variant` won't box</summary>
+
+- `byte`
+- `byte?`
+- `sbyte`
+- `sbyte?`
+- `bool`
+- `bool?`
+- `char`
+- `char?`
+- `short`
+- `short?`
+- `int`
+- `int?`
+- `long`
+- `long?`
+- `ushort`
+- `ushort?`
+- `uint`
+- `uint?`
+- `ulong`
+- `ulong?`
+- `float`
+- `float?`
+- `double`
+- `double?`
+- `DateTimeOffset`
+- `DateTimeOffset?`
+- `DateTime`
+- `DateTime?`
+- Enums
+
+</details>
+
+In principle, `Variant` is similar to `object`, but without boxing as described above.  However, since it's not `object`, there are different APIs needed to achieve some of the same functionality, including:
+
+- [Assigning a value to `Variant`](#assign-a-value-to-variant)
+- [Retrieving the value a `Variant` holds](#get-the-value-from-variant)
+- [Working with `null` and `Variant`](#handling-nulls)
+
+## Assign a value to Variant
+
+`Variant` has implicit cast operators for each of the primitives listed above.  This means you can assign a value of those types to `Variant` without casting, and a new `Variant` instance holding an unboxed copy of the value will be created.
+
+```csharp
+Variant v = 3L;
+
+// v.Type is System.Int64
+```
+
+Since you can't have an implicit cast operator from `object`, for reference types, you must create a new instance of `Variant` and pass the value to the `Variant` constructor.
+
+```csharp
+MemoryStream s = new();
```

Comment:
Do we want to include a `using` statement in this sample?
```suggestion
using MemoryStream s = new();
```

Reply:
Good callout.  I don't want to add complexity to the samples, so I'll pick a type for the examples that isn't disposable to sidestep the issue.  Thanks, @scottaddie!